contracts/TimeLock.sol

```diff
@@ -189,27 +192,35 @@ contract TimeLock is AragonApp, IForwarder, IForwarderFee {
         return (totalAmount, totalDuration);
     }
 
-    function _withdrawTokens(address _sender, uint256 _numberWithdrawLocks) internal {
-        WithdrawLockLib.WithdrawLock[] storage addressWithdrawLocksStorage = addressesWithdrawLocks[_sender];
-        WithdrawLockLib.WithdrawLock[] memory addressWithdrawLocksCopy = addressesWithdrawLocks[_sender];
-
-        require(_numberWithdrawLocks <= addressWithdrawLocksCopy.length, ERROR_TOO_MANY_WITHDRAW_LOCKS);
+    function _withdrawTokens(uint256 _numberWithdrawLocks) internal {
+        WithdrawLock[] storage addressWithdrawLocks = addressesWithdrawLocks[msg.sender];
+        require(_numberWithdrawLocks <= addressWithdrawLocks.length, ERROR_TOO_MANY_WITHDRAW_LOCKS);
 
         uint256 amountOwed = 0;
         uint256 withdrawLockCount = 0;
+        uint256 withdrawLockLength = addressWithdrawLocks.length;
 
-        for (uint256 withdrawLockIndex = 0; withdrawLockIndex < _numberWithdrawLocks; withdrawLockIndex++) {
+        for (uint256 i = _numberWithdrawLocks; i > 0; i--) {
```

Comment:
Hmm, you could use `int256` (or maybe a smaller int). Then I think it could be something like:
```
for (int256 withdrawLockIndex = _numberWithdrawLocks - 1; withdrawLockIndex >= 0; withdrawLockIndex--) {
    ...
}
```
Without the odd `i - 1` inside. Up to you though!

Reply:
I think we would have to  explicitly convert `_numberWithdrawLocks - 1` to `int256` like:   `int256 withdrawLockIndex = int256(_numberWithdrawLocks)` and then when accessing items in the array we would have to convert `withdrawLockIndex`   back again to `uint256` and seems a little bit dangerous. 

Do you know of an alternative to avoid this?